sky/serve/service.py

```diff
@@ -388,7 +387,19 @@ def _get_controller_host():
         for process in process_to_kill:
             process.join()
 
-        failed = _cleanup(service_name, service_spec.pool)
+        # Catch any exception here to avoid it kill the service monitoring
+        # process. In which case, the service will not only fail to clean
+        # up, but also cannot be terminated in the future as no process
+        # will handle the user signal anymore. Instead, we catch any error
+        # and set it to FAILED_CLEANUP instead.
+        try:
+            failed = _cleanup(service_name, service_spec.pool)
+        except Exception as e:  # pylint: disable=broad-except
+            logger.error(f'Failed to clean up service {service_name}: {e}')
+            with ux_utils.enable_traceback():
+                logger.error(f'  Traceback: {traceback.format_exc()}')
```

Comment:
btw, you can use `exc_info=True` parameter to `logger.error`. But this is also fine